components/users/user.js

```diff
@@ -1,32 +1,73 @@
 import mongoose from 'mongoose'
 import bcryptjs from 'bcryptjs'
+import jwt from 'jsonwebtoken'
 
 const userSchema = new mongoose.Schema({
   name: {
     type: String,
-    required: true
+    required: true,
   },
   email: {
     type: String,
     required: true,
-    unique: true
+    unique: true,
   },
   password: {
     type: String,
     required: true,
-    minlength: 8
-  }
+    minlength: 8,
+  },
+  tokens: [
+    {
+      token: {
+        type: String,
+        required: true,
+      },
+    },
+  ],
 })
 
-userSchema.pre('save', async function(next) {
+userSchema.pre('save', async function (next) {
   const user = this
 
-  if(!user.isModified('password')) return
+  if (!user.isModified('password')) return
 
   user.password = await bcryptjs.hash(user.password, 8)
   next()
 })
 
+userSchema.methods.generateToken = async function () {
+  const user = this
+  const secretKey = process.env.SECRET_KEY || 'secret'
```

Comment:
We have a configuration file for `environment variable` let add this line there

Reply:
Fixed